src/commands/openYAMLConfigFile.ts

```diff
@@ -3,35 +3,56 @@
-import { IActionContext } from "vscode-azureextensionui";
+import { basename } from 'path';
+import { Position, Range, TextDocument, window, workspace } from "vscode";
+import { IActionContext, IAzureQuickPickItem } from "vscode-azureextensionui";
 import { ext } from "../extensionVariables";
 import { EnvironmentTreeItem } from "../tree/EnvironmentTreeItem";
+import { BuildConfig, GitHubConfigGroupTreeItem } from '../tree/localProject/ConfigGroupTreeItem';
 import { StaticWebAppTreeItem } from "../tree/StaticWebAppTreeItem";
+import { localize } from '../utils/localize';
 import { openUrl } from "../utils/openUrl";
-import { getSingleRootFsPath } from '../utils/workspaceUtils';
 
-export async function openYAMLConfigFile(context: IActionContext, node?: StaticWebAppTreeItem | EnvironmentTreeItem): Promise<void> {
+export async function openYAMLConfigFile(context: IActionContext, node?: StaticWebAppTreeItem | EnvironmentTreeItem | GitHubConfigGroupTreeItem, buildConfigToSelect?: BuildConfig): Promise<void> {
     if (!node) {
         node = await ext.tree.showTreeItemPicker<EnvironmentTreeItem>(EnvironmentTreeItem.contextValue, context);
     }
 
-    const defaultHostname: string = node instanceof StaticWebAppTreeItem ? node.defaultHostname : node.parent.defaultHostname;
-    const ymlFileName: string = `.github/workflows/azure-static-web-apps-${defaultHostname.split('.')[0]}.yml`;
-
-    if (node instanceof EnvironmentTreeItem && node.inWorkspace) {
-        const fsPath: string | undefined = getSingleRootFsPath();
-        if (fsPath) {
-            const ymlFsPath: string = path.join(fsPath, ymlFileName);
-            // if we couldn't find it, then try opening it in GitHub
-            if (await fse.pathExists(ymlFsPath)) {
-                await window.showTextDocument(await workspace.openTextDocument(ymlFsPath));
-                return;
-            }
+    if (node instanceof StaticWebAppTreeItem || node instanceof EnvironmentTreeItem && node.gitHubConfigGroupTreeItems.length === 0) {
+        const defaultHostname: string = node instanceof StaticWebAppTreeItem ? node.defaultHostname : node.parent.defaultHostname;
+        const ymlFileName: string = `.github/workflows/azure-static-web-apps-${defaultHostname.split('.')[0]}.yml`;
+        return await openUrl(`${node.repositoryUrl}/edit/${node.branch}/${ymlFileName}`);
+    }
+
+    let yamlFilePath: string | undefined;
+    if (node instanceof GitHubConfigGroupTreeItem ){
+        yamlFilePath = node.yamlFilePath;
+    } else {
+        const picks: IAzureQuickPickItem<string>[] = node.gitHubConfigGroupTreeItems.map(configNode => {
+            return { label: basename(configNode.yamlFilePath), data: configNode.yamlFilePath };
+        });
+
+        if (picks.length === 1) {
+            yamlFilePath = picks[0].data;
+        } else {
+            const placeHolder: string = localize('selectGitHubConfig', 'Select the GitHub configuration file to open.');
```

Comment:
I think we should call this a GitHub workflow file.  It's what the bot calls it when it commits the file:

![image](https://user-images.githubusercontent.com/5290572/113622401-c9d93d80-9611-11eb-9e89-ae69af7118b2.png)

The configuration portion of it is what we're looking at, but the entire thing is a workflow file and lives in the .github/.workflows folder.